Tests fail due to C++ compilation error

I checked out the repo and ran `cargo +nightly test`. Two tests failed (output below). `g++ --version` says it's 5.5.0-12ubunutu1~16.04)

```
failures:

---- test_associated_in_body stdout ----
Running: "/home/kats/zspace/cbindgen/target/debug/cbindgen" "--lang" "c" "--cpp-compat" "--style" "type" "-o" "/home/kats/zspace/cbindgen/tests/expectations/associated_in_body.compat.c" "--config" "/home/kats/zspace/cbindgen/tests/rust/associated_in_body.toml" "/home/kats/zspace/cbindgen/tests/rust/associated_in_body.rs"
Running: "gcc" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/associated_in_body.compat.c" "-o" "/home/kats/tmp/cbindgen-test-outputiRwLsY/associated_in_body.compat.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-D" "CBINDGEN_STYLE_TYPE"
Running: "g++" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/associated_in_body.compat.c" "-o" "/home/kats/tmp/cbindgen-test-outputiRwLsY/associated_in_body.compat.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-std=c++17" "-D" "CBINDGEN_STYLE_TYPE"
Running: "/home/kats/zspace/cbindgen/target/debug/cbindgen" "--lang" "c" "--style" "type" "-o" "/home/kats/zspace/cbindgen/tests/expectations/associated_in_body.c" "--config" "/home/kats/zspace/cbindgen/tests/rust/associated_in_body.toml" "/home/kats/zspace/cbindgen/tests/rust/associated_in_body.rs"
Running: "gcc" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/associated_in_body.c" "-o" "/home/kats/tmp/cbindgen-test-outputiRwLsY/associated_in_body.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-D" "CBINDGEN_STYLE_TYPE"
Running: "/home/kats/zspace/cbindgen/target/debug/cbindgen" "--lang" "c" "--cpp-compat" "--style" "tag" "-o" "/home/kats/zspace/cbindgen/tests/expectations/tag/associated_in_body.compat.c" "--config" "/home/kats/zspace/cbindgen/tests/rust/associated_in_body.toml" "/home/kats/zspace/cbindgen/tests/rust/associated_in_body.rs"
Running: "gcc" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/tag/associated_in_body.compat.c" "-o" "/home/kats/tmp/cbindgen-test-outputiRwLsY/associated_in_body.compat.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-D" "CBINDGEN_STYLE_TAG"
Running: "g++" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/tag/associated_in_body.compat.c" "-o" "/home/kats/tmp/cbindgen-test-outputiRwLsY/associated_in_body.compat.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-std=c++17" "-D" "CBINDGEN_STYLE_TAG"
Running: "/home/kats/zspace/cbindgen/target/debug/cbindgen" "--lang" "c" "--style" "tag" "-o" "/home/kats/zspace/cbindgen/tests/expectations/tag/associated_in_body.c" "--config" "/home/kats/zspace/cbindgen/tests/rust/associated_in_body.toml" "/home/kats/zspace/cbindgen/tests/rust/associated_in_body.rs"
Running: "gcc" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/tag/associated_in_body.c" "-o" "/home/kats/tmp/cbindgen-test-outputiRwLsY/associated_in_body.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-D" "CBINDGEN_STYLE_TAG"
Running: "/home/kats/zspace/cbindgen/target/debug/cbindgen" "--lang" "c" "--cpp-compat" "--style" "both" "-o" "/home/kats/zspace/cbindgen/tests/expectations/both/associated_in_body.compat.c" "--config" "/home/kats/zspace/cbindgen/tests/rust/associated_in_body.toml" "/home/kats/zspace/cbindgen/tests/rust/associated_in_body.rs"
Running: "gcc" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/both/associated_in_body.compat.c" "-o" "/home/kats/tmp/cbindgen-test-outputiRwLsY/associated_in_body.compat.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-D" "CBINDGEN_STYLE_BOTH"
Running: "g++" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/both/associated_in_body.compat.c" "-o" "/home/kats/tmp/cbindgen-test-outputiRwLsY/associated_in_body.compat.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-std=c++17" "-D" "CBINDGEN_STYLE_BOTH"
Running: "/home/kats/zspace/cbindgen/target/debug/cbindgen" "--lang" "c" "--style" "both" "-o" "/home/kats/zspace/cbindgen/tests/expectations/both/associated_in_body.c" "--config" "/home/kats/zspace/cbindgen/tests/rust/associated_in_body.toml" "/home/kats/zspace/cbindgen/tests/rust/associated_in_body.rs"
Running: "gcc" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/both/associated_in_body.c" "-o" "/home/kats/tmp/cbindgen-test-outputiRwLsY/associated_in_body.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-D" "CBINDGEN_STYLE_BOTH"
Running: "/home/kats/zspace/cbindgen/target/debug/cbindgen" "-o" "/home/kats/zspace/cbindgen/tests/expectations/associated_in_body.cpp" "--config" "/home/kats/zspace/cbindgen/tests/rust/associated_in_body.toml" "/home/kats/zspace/cbindgen/tests/rust/associated_in_body.rs"
Running: "g++" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/associated_in_body.cpp" "-o" "/home/kats/tmp/cbindgen-test-outputiRwLsY/associated_in_body.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-std=c++17"
thread 'test_associated_in_body' panicked at 'Output failed to compile: Output { status: ExitStatus(ExitStatus(256)), stdout: "", stderr: "/home/kats/zspace/cbindgen/tests/expectations/associated_in_body.cpp:46:47: error: ‘StyleAlignFlags::AUTO’ declared as an ‘inline’ variable\n inline const StyleAlignFlags StyleAlignFlags::AUTO = StyleAlignFlags{ /* .bits = */ 0 };\n                                               ^\n/home/kats/zspace/cbindgen/tests/expectations/associated_in_body.cpp:48:47: error: ‘StyleAlignFlags::NORMAL’ declared as an ‘inline’ variable\n inline const StyleAlignFlags StyleAlignFlags::NORMAL = StyleAlignFlags{ /* .bits = */ 1 };\n                                               ^\n/home/kats/zspace/cbindgen/tests/expectations/associated_in_body.cpp:50:47: error: ‘StyleAlignFlags::START’ declared as an ‘inline’ variable\n inline const StyleAlignFlags StyleAlignFlags::START = StyleAlignFlags{ /* .bits = */ (1 << 1) };\n                                               ^\n/home/kats/zspace/cbindgen/tests/expectations/associated_in_body.cpp:52:47: error: ‘StyleAlignFlags::END’ declared as an ‘inline’ variable\n inline const StyleAlignFlags StyleAlignFlags::END = StyleAlignFlags{ /* .bits = */ (1 << 2) };\n                                               ^\n/home/kats/zspace/cbindgen/tests/expectations/associated_in_body.cpp:54:47: error: ‘StyleAlignFlags::FLEX_START’ declared as an ‘inline’ variable\n inline const StyleAlignFlags StyleAlignFlags::FLEX_START = StyleAlignFlags{ /* .bits = */ (1 << 3) };\n                                               ^\n" }', tests/tests.rs:118:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- test_constant_constexpr stdout ----
Running: "/home/kats/zspace/cbindgen/target/debug/cbindgen" "--lang" "c" "--cpp-compat" "--style" "type" "-o" "/home/kats/zspace/cbindgen/tests/expectations/constant_constexpr.compat.c" "--config" "/home/kats/zspace/cbindgen/tests/rust/constant_constexpr.toml" "/home/kats/zspace/cbindgen/tests/rust/constant_constexpr.rs"
Running: "gcc" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/constant_constexpr.compat.c" "-o" "/home/kats/tmp/cbindgen-test-outputTOOZtd/constant_constexpr.compat.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-D" "CBINDGEN_STYLE_TYPE"
Running: "g++" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/constant_constexpr.compat.c" "-o" "/home/kats/tmp/cbindgen-test-outputTOOZtd/constant_constexpr.compat.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-std=c++17" "-D" "CBINDGEN_STYLE_TYPE"
Running: "/home/kats/zspace/cbindgen/target/debug/cbindgen" "--lang" "c" "--style" "type" "-o" "/home/kats/zspace/cbindgen/tests/expectations/constant_constexpr.c" "--config" "/home/kats/zspace/cbindgen/tests/rust/constant_constexpr.toml" "/home/kats/zspace/cbindgen/tests/rust/constant_constexpr.rs"
Running: "gcc" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/constant_constexpr.c" "-o" "/home/kats/tmp/cbindgen-test-outputTOOZtd/constant_constexpr.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-D" "CBINDGEN_STYLE_TYPE"
Running: "/home/kats/zspace/cbindgen/target/debug/cbindgen" "--lang" "c" "--cpp-compat" "--style" "tag" "-o" "/home/kats/zspace/cbindgen/tests/expectations/tag/constant_constexpr.compat.c" "--config" "/home/kats/zspace/cbindgen/tests/rust/constant_constexpr.toml" "/home/kats/zspace/cbindgen/tests/rust/constant_constexpr.rs"
Running: "gcc" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/tag/constant_constexpr.compat.c" "-o" "/home/kats/tmp/cbindgen-test-outputTOOZtd/constant_constexpr.compat.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-D" "CBINDGEN_STYLE_TAG"
Running: "g++" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/tag/constant_constexpr.compat.c" "-o" "/home/kats/tmp/cbindgen-test-outputTOOZtd/constant_constexpr.compat.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-std=c++17" "-D" "CBINDGEN_STYLE_TAG"
Running: "/home/kats/zspace/cbindgen/target/debug/cbindgen" "--lang" "c" "--style" "tag" "-o" "/home/kats/zspace/cbindgen/tests/expectations/tag/constant_constexpr.c" "--config" "/home/kats/zspace/cbindgen/tests/rust/constant_constexpr.toml" "/home/kats/zspace/cbindgen/tests/rust/constant_constexpr.rs"
Running: "gcc" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/tag/constant_constexpr.c" "-o" "/home/kats/tmp/cbindgen-test-outputTOOZtd/constant_constexpr.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-D" "CBINDGEN_STYLE_TAG"
Running: "/home/kats/zspace/cbindgen/target/debug/cbindgen" "--lang" "c" "--cpp-compat" "--style" "both" "-o" "/home/kats/zspace/cbindgen/tests/expectations/both/constant_constexpr.compat.c" "--config" "/home/kats/zspace/cbindgen/tests/rust/constant_constexpr.toml" "/home/kats/zspace/cbindgen/tests/rust/constant_constexpr.rs"
Running: "gcc" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/both/constant_constexpr.compat.c" "-o" "/home/kats/tmp/cbindgen-test-outputTOOZtd/constant_constexpr.compat.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-D" "CBINDGEN_STYLE_BOTH"
Running: "g++" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/both/constant_constexpr.compat.c" "-o" "/home/kats/tmp/cbindgen-test-outputTOOZtd/constant_constexpr.compat.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-std=c++17" "-D" "CBINDGEN_STYLE_BOTH"
Running: "/home/kats/zspace/cbindgen/target/debug/cbindgen" "--lang" "c" "--style" "both" "-o" "/home/kats/zspace/cbindgen/tests/expectations/both/constant_constexpr.c" "--config" "/home/kats/zspace/cbindgen/tests/rust/constant_constexpr.toml" "/home/kats/zspace/cbindgen/tests/rust/constant_constexpr.rs"
Running: "gcc" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/both/constant_constexpr.c" "-o" "/home/kats/tmp/cbindgen-test-outputTOOZtd/constant_constexpr.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-D" "CBINDGEN_STYLE_BOTH"
Running: "/home/kats/zspace/cbindgen/target/debug/cbindgen" "-o" "/home/kats/zspace/cbindgen/tests/expectations/constant_constexpr.cpp" "--config" "/home/kats/zspace/cbindgen/tests/rust/constant_constexpr.toml" "/home/kats/zspace/cbindgen/tests/rust/constant_constexpr.rs"
Running: "g++" "-D" "DEFINED" "-c" "/home/kats/zspace/cbindgen/tests/expectations/constant_constexpr.cpp" "-o" "/home/kats/tmp/cbindgen-test-outputTOOZtd/constant_constexpr.o" "-I" "/home/kats/zspace/cbindgen/tests" "-Wall" "-Werror" "-Wno-attributes" "-std=c++17"
thread 'test_constant_constexpr' panicked at 'Output failed to compile: Output { status: ExitStatus(ExitStatus(256)), stdout: "", stderr: "/home/kats/zspace/cbindgen/tests/expectations/constant_constexpr.cpp:18:37: error: ‘Foo::CONSTANT_I64_BODY’ declared as an ‘inline’ variable\n constexpr inline const int64_t Foo::CONSTANT_I64_BODY = 216;\n                                     ^\n" }', tests/tests.rs:118:5


failures:
    test_associated_in_body
    test_constant_constexpr

test result: FAILED. 104 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out
```

--- a/tests/tests.rs
+++ b/tests/tests.rs
@@ -83,8 +83,6 @@ fn compile(
 
     let mut command = Command::new(cc);
     command.arg("-D").arg("DEFINED");
-    command.arg("-c").arg(cbindgen_output);
-    command.arg("-o").arg(&object);
     command.arg("-I").arg(tests_path);
     command.arg("-Wall");
     if !skip_warning_as_error {
@@ -92,10 +90,17 @@ fn compile(
     }
     // `swift_name` is not recognzied by gcc.
     command.arg("-Wno-attributes");
+    // clang warns about unused const variables.
+    command.arg("-Wno-unused-const-variable");
+    // clang also warns about returning non-instantiated templates (they could
+    // be specialized, but they're not so it's fine).
+    command.arg("-Wno-return-type-c-linkage");
     if let Language::Cxx = language {
         // enum class is a c++11 extension which makes g++ on macos 10.14 error out
         // inline variables are are a c++17 extension
         command.arg("-std=c++17");
+        // Prevents warnings when compiling .c files as c++.
+        command.arg("-x").arg("c++");
         if let Ok(extra_flags) = env::var("CXXFLAGS") {
             command.args(extra_flags.split_whitespace());
         }
@@ -112,6 +117,9 @@ fn compile(
             style_str(style).to_uppercase()
         ));
     }
+
+    command.arg("-o").arg(&object);
+    command.arg("-c").arg(cbindgen_output);
 
     println!("Running: {:?}", command);
     let out = command.output().expect("failed to compile");

